tools/genconversion/conversion.go

```diff
@@ -1,90 +1,56 @@
+// conversion-gen is a tool for auto-generating Conversion functions.
+//
+// Structs in the input directories with the below line in their comments
+// will be ignored during generation.
+// // +genconversion=false
 package main
 
 import (
-	"fmt"
-	"io"
-	"log"
-	"os"
-	"runtime"
-	"sort"
-	"strings"
-
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/api/unversioned"
-	pkg_runtime "k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/cmd/libs/go2idl/args"
+	"k8s.io/kubernetes/cmd/libs/go2idl/conversion-gen/generators"
 
 	"github.com/golang/glog"
-	flag "github.com/spf13/pflag"
-
-	_ "github.com/openshift/origin/pkg/api"
-	_ "github.com/openshift/origin/pkg/api/v1"
-	_ "github.com/openshift/origin/pkg/api/v1beta3"
-
-	// install all APIs
-	_ "github.com/openshift/origin/pkg/api/install"
-	_ "k8s.io/kubernetes/pkg/api/install"
-	_ "k8s.io/kubernetes/pkg/apis/extensions/install"
-)
-
-var (
-	functionDest = flag.StringP("funcDest", "f", "-", "Output for conversion functions; '-' means stdout")
-	group        = flag.StringP("group", "g", "", "Group for conversion.")
-	version      = flag.StringP("version", "v", "v1beta3", "Version for conversion.")
 )
 
 func main() {
-	runtime.GOMAXPROCS(runtime.NumCPU())
-	flag.Parse()
-	log.SetOutput(os.Stderr)
-
-	var funcOut io.Writer
-	if *functionDest == "-" {
-		funcOut = os.Stdout
-	} else {
-		file, err := os.Create(*functionDest)
-		if err != nil {
-			glog.Fatalf("Couldn't open %v: %v", *functionDest, err)
-		}
-		defer file.Close()
-		funcOut = file
+	arguments := args.Default()
+
+	// Override defaults. These are Kubernetes specific input locations.
+	arguments.InputDirs = []string{
```

Comment:
We should either auto-detect or write a test to make sure we aren't missing any.  I think we could do it by walking the scheme and doing a series of reflective checks.